src/parser/jobs/pld/modules/RoyalAuthority.tsx

```diff
@@ -0,0 +1,193 @@
+			action: this.data.actions.HOLY_SPIRIT.id,
+		}, this.onHolySpirit)
+
+		this.addEventHook({
+			type: 'action',
+			source: this.parser.actor.id,
+			action: this.data.actions.HOLY_CIRCLE.id,
+		}, this.onHolyCircle)
+
+		this.addEventHook({
+			type: 'statusApply',
+			target: this.parser.actor.id,
+			status: this.data.statuses.SWORD_OATH.id,
+		}, this.onApplySwordOath)
+
+		this.addEventHook({
+			type: 'statusRemove',
+			target: this.parser.actor.id,
+			status: this.data.statuses.SWORD_OATH.id,
+		}, this.onRemoveSwordOath)
+
+		this.addEventHook({
+			type: 'statusApply',
+			target: this.parser.actor.id,
+			status: this.data.statuses.DIVINE_MIGHT.id,
+		}, this.onApplyDivineMight)
+
+		this.addEventHook({
+			type: 'statusRemove',
+			target: this.parser.actor.id,
+			status: this.data.statuses.DIVINE_MIGHT.id,
+		}, this.onRemoveDivineMight)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onAtonement() {
+		if (this.currentSwordOath == null) { return }
+		this.currentSwordOath.used++
+	}
+
+	private onHolySpirit() {
+		if (this.divineMightActive === false) { return }
+		this.divineMightUses++
+	}
+
+	private onHolyCircle() {
+		if (this.divineMightActive === false) { return }
+		this.divineMightUses++
+	}
+
+	private onApplySwordOath(event: Events['statusApply']): void {
+		if (event.data == null) { return }
+
+		// Track potential uses & any overcap due to reapplication while the status was still active
+		if (event.data === this.data.statuses.SWORD_OATH.stacksApplied) {
+			if (this.currentSwordOath != null) {
+				this.swordOathOvercap += this.currentSwordOath.stacks
+				this.onRemoveSwordOath()
+			}
+		}
+
+		if (this.currentSwordOath == null) {
+			this.currentSwordOath = {
+				initial: event.data,
+				stacks: event.data,
+				used: 0,
+			}
+		} else {
+			this.currentSwordOath.stacks = event.data
+		}
+	}
+
+	private onRemoveSwordOath(): void {
+		if (this.currentSwordOath == null) { return }
+		this.swordOathHistory.push(this.currentSwordOath)
+		this.currentSwordOath = undefined
+	}
+
+	private onApplyDivineMight(): void {
+		this.divineMightActive = true
+		this.divineMightTotal++
+	}
+
+	private onRemoveDivineMight(): void {
+		this.divineMightActive = false
+	}
+
+	private onComplete() {
+		this.onRemoveSwordOath()
+		this.onRemoveDivineMight()
+
+		const swordOathStacks = this.data.statuses.SWORD_OATH.stacksApplied
+		const swordOathStacksUsed = this.swordOathHistory.reduce((used, swordOath) => used + swordOath.used, 0)
+		const swordOathPotentialStacks = this.swordOathHistory.reduce((potential, swordOath) => potential + swordOath.initial, 0)
+		// Leaving the following commented out since it's not actually being used right now, but here's the calculation for checking dropped stacks if that data becomes useful
+		// const droppedStacks = this.swordOathHistory.reduce((dropped, swordOath) => dropped + Math.max(swordOath.initial - swordOath.used, 0), 0)
+
+		this.checklist.add(new TieredRule({
```

Comment:
i'm not super on top of the current recommendations for pld; i've seen some rotations with additional hardcasts of hs, and/or dropped atonements, for alignment purposes.

are they recommended enough to care about?
and if they are, is this checklist going to start dinging people who drop atonements for that purpose?
and if it _is_ - is it niche enough to not worry about anyway?
genuine questions, idk

Reply:
No, hardcasts of hs are not recommended because they are damage losses.

Reply:
![image](https://user-images.githubusercontent.com/10051083/213924648-bc37e0b9-0d33-4027-83a9-e2ae1c24d152.png)

Reply:
Yeah, had read that. The thing of note is the "almost exactly the same" bit - _iif_ the looping rotations are common _enough_ to ding people regularly for dropped atonements, then we'll probably need to look at i.e. reducing the breakpoint. But for now, it should be fine.